Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -3404,22 +3460,31 @@ - (void)_hideFindOverlay
         if (!strongSelf)
             return;
 
-        if ([strongSelf->_findOverlay alpha])
-            return;
+        [strongSelf _removeLayerForFindOverlay];
 
-        [strongSelf->_findOverlay removeFromSuperview];
-        strongSelf->_findOverlay = nil;
+        if (!strongSelf->_findOverlaysOutsideContentView || [strongSelf->_findOverlaysOutsideContentView->top alpha])
+            return;
 
-        [strongSelf _removeLayerForFindOverlay];
+        [strongSelf->_findOverlaysOutsideContentView->top removeFromSuperview];
+        [strongSelf->_findOverlaysOutsideContentView->bottom removeFromSuperview];
+        [strongSelf->_findOverlaysOutsideContentView->left removeFromSuperview];
+        [strongSelf->_findOverlaysOutsideContentView->right removeFromSuperview];
+        strongSelf->_findOverlaysOutsideContentView.reset();
     }];
 
     [contentViewFindOverlayLayer addAnimation:animation forKey:@"findOverlayFadeOut"];
-    [findOverlayLayer addAnimation:animation forKey:@"findOverlayFadeOut"];
+    [[_findOverlaysOutsideContentView.value().top layer] addAnimation:animation forKey:@"findOverlayFadeOut"];
+    [[_findOverlaysOutsideContentView.value().bottom layer] addAnimation:animation forKey:@"findOverlayFadeOut"];
+    [[_findOverlaysOutsideContentView.value().left layer] addAnimation:animation forKey:@"findOverlayFadeOut"];
+    [[_findOverlaysOutsideContentView.value().right layer] addAnimation:animation forKey:@"findOverlayFadeOut"];
 
     [CATransaction commit];
 
     contentViewFindOverlayLayer.opacity = 0;
-    findOverlayLayer.opacity = 0;
+    [_findOverlaysOutsideContentView.value().top layer].opacity = 0;
```

Comment:
Anywhere you use `value()` should just be `->`.

But please also try to do this https://github.com/WebKit/WebKit/pull/25600#discussion_r1516993930, so that only needs to happen in one place.